src/Perlang.Parser/Scanner.cs

```diff
@@ -356,6 +356,9 @@ private void Number()
 
             if (isFractional)
             {
+                // TODO: This is a mess. We currently treat all floating point values as _double_, which is insane. We
+                // TODO: should probably have a "use smallest possible type" logic as below for integers, for flotaing point
+                // TODO: values as well. We could also consider supporting `decimal` while we're at it.
```

Comment:
See #70. I think that adding an explicit `float` and `double` should get fairly high priority, since the current way (of just putting everything in there as `double`s) is extremely messed up, to be honest.